From 0f931616cbf61aecc38cf06c945b15ebd63e5de1 Mon Sep 17 00:00:00 2001
From: Alona Solntseva <alona.solntseva@tandemg.com>
Date: Wed, 09 Feb 2022 18:22:41 +0200
Subject: [PATCH] [SW-67906] erom: rebuild erom images

Rebuild erom images for erom to allow compilation separation inside a
file

Change-Id: I8b16f22f598271f69b65967b6247761ccdd4d5e9
Signed-off-by: Alona Solntseva <alona.solntseva@tandemg.com>
---

diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index 0b97d4c..6a80d33 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -12,7 +12,8 @@
 CFLAGS += -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg
 CFLAGS += -Wall -Werror
 CFLAGS += -g -c
-CFLAGS += -I $(PREBOOT_ROOT_DIR)/include/ -I $(PREBOOT_ROOT_DIR)/ -I $(PREBOOT_ROOT_DIR)/lib/common/ -I $(HL_DEV_ARCH)/
+CFLAGS += -I $(PREBOOT_ROOT_DIR)/include/ -I $(PREBOOT_ROOT_DIR)/ -I $(PREBOOT_ROOT_DIR)/lib/common/ -I $(PREBOOT_ROOT_DIR)/lib/
+CFLAGS += -I $(HL_DEV_ARCH)/ -I $(HL_DEV_ARCH)/../common/ -I $(SPECS_EMBEDDED_DIR)/$(BASE_DEV_NAME)/
 CFLAGS += -include $(SPECS_EMBEDDED_DIR)/$(SPECS_DEV)/$(SPECS_DEV)_reg_map.h
 
 CFLAGS += -Os -fdata-sections -ffunction-sections
@@ -62,7 +63,10 @@
 EROM_RECOVERY_EXTRA_LIBS := "-lfdt"
 
 .PHONY: all
-all: erom_recovery erom_pcie erom_tpm
+all: clean
+	$(MAKE) erom_recovery
+	$(MAKE) erom_pcie
+	$(MAKE) erom_tpm
 
 erom_recovery: erom_recovery.o $(EROM_RECOVERY_DEPS) $(EROM_APP_OBJECTS) $(EROM_RECOVERY_LD_SCRIPT)
 	@$(CC) $(LDFLAGS) -T $(EROM_RECOVERY_LD_SCRIPT) -o $@  -Wl,--start-group $< $(EROM_RECOVERY_DEPS) $(LDLIBS) $(EROM_RECOVERY_EXTRA_LIBS) -Wl,--end-group
@@ -86,6 +90,6 @@
 	@$(CC) $(SFLAGS) -o $@ $<
 
 clean:
-	rm -f *.o
+	rm -f $(EROM_APP_OBJECTS) $(EROM_TPM_DEPS) $(EROM_RECOVERY_DEPS) $(EROM_PCIE_DEPS)
 
 .PHONY: clean
